### ISSUE TITLE
muffled audio output for custom dataset

### ISSUE THREAD

**Issue by reubzdubz** [NVIDIA/radtts#7]
I have followed your work with Flowtron and it produces audio really nicely. I am using the same dataset of 1.5 hrs of a single speaker but I have trouble getting some output that is not as muffled here. The alignment is decent:

![Screenshot 2022-07-21 182200](https://user-images.githubusercontent.com/42710108/180191653-4d264aca-f8fe-4a0e-b607-502ff9064a76.png)

 and I am using the following parameters, am I missing something? Or is it just because I am training from scratch (I already did the decoder step)? Any help will be appreciated, thanks!

{
    "train_config": {
        "output_directory": "/debug",
        "epochs": 10000000,
        "optim_algo": "RAdam",
        "learning_rate": 1e-4,
        "weight_decay": 1e-6,
        "sigma": 1.0,
         "iters_per_checkpoint": 185,
         "batch_size": 4,
        "seed": null,
        "checkpoint_path": "",
        "ignore_layers": [],
        "ignore_layers_warmstart": [],
        "finetune_layers": [],
        "include_layers": [],
         "vocoder_config_path": "/content/drive/MyDrive/hifigan_22khz_config.json",
         "vocoder_checkpoint_path": "/content/drive/MyDrive/hifigan_libritts100360_generator0p5.pt",
        "log_attribute_samples": true,
        "log_decoder_samples": true,

        "warmstart_checkpoint_path": "path/to/pretrained/decoder",
        "use_amp": false,
        "grad_clip_val": 1.0,
        "loss_weights": {
            "blank_logprob": -1, 
            "ctc_loss_weight": 0.1,
            "binarization_loss_weight": 1.0,
            "dur_loss_weight": 1.0,
            "f0_loss_weight": 1.0,
            "energy_loss_weight": 1.0,
            "vpred_loss_weight": 1.0
        },
        "binarization_start_iter": 0,
        "kl_loss_start_iter": 0,
        "unfreeze_modules": "durf0energyvpred"
    },
    "data_config": {
        "training_files": {
            "LJS": {
                "basedir": "",
                "audiodir": "",
                "filelist":  "/content/drive/MyDrive/bastila_wav/filelists/bastila_train_rad.txt",
                "lmdbpath": ""
            }
        },
        "validation_files": {
            "LJS": {
                "basedir": "",
                "audiodir": "",
                "filelist":  "/content/drive/MyDrive/bastila_wav/filelists/bastila_val_rad.txt",
                "lmdbpath": ""
            }
        },
        "dur_min": 0.1,
        "dur_max": 10.2,
        "sampling_rate": 22050,
        "filter_length": 1024,
        "hop_length": 256,
        "win_length": 1024,
        "n_mel_channels": 80,
        "mel_fmin": 0.0,
        "mel_fmax": 8000.0,
        "f0_min": 80.0,
        "f0_max": 640.0,
        "max_wav_value": 32768.0,
        "use_f0": true,
        "use_log_f0": 0,
        "use_energy_avg": true,
        "use_scaled_energy": true,
        "symbol_set": "radtts",
        "cleaner_names": ["radtts_cleaners"],
        "heteronyms_path": "tts_text_processing/heteronyms",
        "phoneme_dict_path": "tts_text_processing/cmudict-0.7b",
        "p_phoneme": 1.0,
        "handle_phoneme": "word",
        "handle_phoneme_ambiguous": "ignore",
        "include_speakers": null,
        "n_frames": -1,
        "betabinom_cache_path": "data_cache/",
        "lmdb_cache_path": "", 
        "use_attn_prior_masking": true,
        "prepend_space_to_text": true,
        "append_space_to_text": true,
        "add_bos_eos_to_text": false,
        "betabinom_scaling_factor": 1.0,
        "distance_tx_unvoiced": false,
        "mel_noise_scale": 0.0
    },
    "dist_config": {
        "dist_backend": "nccl",
        "dist_url": "tcp://localhost:54321"
    },
    "model_config": {
        "n_speakers": 1,
        "n_speaker_dim": 16,
        "n_text": 185,
        "n_text_dim": 512,
        "n_flows": 8,
        "n_conv_layers_per_step": 4,
        "n_mel_channels": 80,
        "n_hidden": 1024,
        "mel_encoder_n_hidden": 512,
        "dummy_speaker_embedding": false,
        "n_early_size": 2,
        "n_early_every": 2,
        "n_group_size": 2,
        "affine_model": "wavenet",
        "include_modules": "decatndpmvpredapm",
        "scaling_fn": "tanh",
        "matrix_decomposition": "LUS",
        "learn_alignments": true,
        "use_speaker_emb_for_alignment": false,
        "attn_straight_through_estimator": true,
        "use_context_lstm": true,
        "context_lstm_norm": "spectral",
        "context_lstm_w_f0_and_energy": true,
        "text_encoder_lstm_norm": "spectral",
        "n_f0_dims": 1,
        "n_energy_avg_dims": 1,
        "use_first_order_features": false,
        "unvoiced_bias_activation": "relu",
        "decoder_use_partial_padding": true,
        "decoder_use_unvoiced_bias": true,
        "ap_pred_log_f0": true,
        "ap_use_unvoiced_bias": true,
        "ap_use_voiced_embeddings": true,
        "dur_model_config": {
            "name": "dap",
            "hparams": {
                "n_speaker_dim": 16,
                "bottleneck_hparams": {
                    "in_dim": 512,
                    "reduction_factor": 16,
                    "norm": "weightnorm",
                    "non_linearity": "relu"
                },
                "take_log_of_input": true,
                "arch_hparams": {
                    "out_dim": 1,
                    "n_layers": 2,
                    "n_channels": 256,
                    "kernel_size": 3,
                    "p_dropout": 0.25
                }
            }
        },
        "f0_model_config": {
            "name": "agap",
            "hparams": {
                "n_in_dim": 1,
                "n_group_size": 1,
                "take_log_of_input": false,
                "n_speaker_dim": 22,
                "n_flows": 2,
                "n_hidden": 128,
                "n_lstm_layers": 1,
                "scaling_fn": "tanh",
                "bottleneck_hparams": {
                    "in_dim": 512,
                    "reduction_factor": 16,
                    "norm": "weightnorm",
                    "non_linearity": "relu"
                },
                "spline_flow_params": {
                    "n_in_channels": 1,
                    "n_context_dim": 128,
                    "n_layers": 4,
                    "n_bins": 24,
                    "use_quadratic": true
                }
            }
        },
        "energy_model_config": {
            "name": "agap",
            "hparams": {
                "n_in_dim": 1,
                "n_group_size": 1,
                "take_log_of_input": false,
                "n_speaker_dim": 22,
                "n_flows": 4,
                "n_hidden": 128,
                "n_lstm_layers": 1,
                "scaling_fn": "tanh",
                "bottleneck_hparams": {
                    "in_dim": 512,
                    "reduction_factor": 16,
                    "norm": "weightnorm",
                    "non_linearity": "relu"
                },
                "spline_flow_params": {
                    "n_in_channels": 1,
                    "n_context_dim": 128,
                    "n_layers": 4,
                    "n_bins": 24,
                    "use_quadratic": true
                }
            }
        },
        "v_model_config": {
            "name": "dap",
            "hparams": {
                "n_speaker_dim": 16,
                "take_log_of_input": false,
                "bottleneck_hparams": {
                    "in_dim": 512,
                    "reduction_factor": 16,
                    "norm": "weightnorm",
                    "non_linearity": "relu"
                },
                "arch_hparams": {
                    "out_dim": 1,
                    "n_layers": 2,
                    "n_channels": 256,
                    "kernel_size": 3,
                    "p_dropout": 0.5,
                    "lstm_type": "",
                    "use_linear": 1
                }
            }
        }
    }
}

**Comment by rafaelvalle** [comment]
Can you share tensorboard pictures and decoder samples using ground truth features?

**Comment by reubzdubz** [comment]
Here are the tensorboard features after training the decoder step on 17 epochs:
![Screenshot_2022-07-22-06-50-00-57_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339225-163c7e8e-d1d4-43fa-b2ad-c2665b98230b.jpg)
![Screenshot_2022-07-22-07-34-31-17_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339256-a5d8bd3a-d9e2-4590-9545-adebce64769b.jpg)
![Screenshot_2022-07-22-07-34-39-62_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339260-35dfcd9d-b4aa-43c9-a44f-a8d835337618.jpg)

And the validation losses after training it using agap:
![Screenshot_2022-07-22-08-41-39-25_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339505-7b5ee204-df0d-4f19-82de-10bba7bc63e6.jpg)
![Screenshot_2022-07-22-08-41-46-30_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339512-489b7da3-5eb3-486c-816b-e6461023a29b.jpg)
![Screenshot_2022-07-22-08-41-57-68_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339517-49c7c6f2-c8cd-4b6e-84a4-1f873ba74670.jpg)
![Screenshot_2022-07-22-08-43-13-25_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339522-889066a7-5bee-40b4-bd77-d6a8b4a175b1.jpg)
![Screenshot_2022-07-22-09-00-28-58_4641ebc0df1485bf6b47ebd018b5ee76](https://user-images.githubusercontent.com/42710108/180339525-b73160a5-ca4e-4707-bffc-f04041d495a1.jpg)


As for the ground truth features are you referring to this?

https://user-images.githubusercontent.com/42710108/180339718-6f9f6e4d-5273-4c77-b886-4553a2929d13.mp4

**Comment by rafaelvalle** [comment]
Yes, please do so.

**Comment by reubzdubz** [comment]
https://colab.research.google.com/drive/16wddBHEBYidtk45lquEmeEUtNcXDONlv

Here is the link

**Comment by rafaelvalle** [comment]
```! python3 train.py -c configs/config_ljs_agap.json -p train_config.learning_rate=0.0005 model_config.n_speakers=1  model_config.n_flows=4 train_config.warmstart_checkpoint_path="/content/drive/MyDrive/flowtron_checkpoints/radtts_test2" train_config.output_directory="/content/drive/MyDrive/flowtron_checkpoints/"```

`configs/config_ljs_agap.json` by default only trains the attribute predictors, not the decoder, so you're not updating the timbre to match your speaker's, just the prosody.

if you want train both the decoder and the feature predictor at the same time you'll have to set [train_config.unfreeze_modules="all"](https://github.com/NVIDIA/radtts/blob/main/configs/config_ljs_decoder.json#L35)

**Comment by reubzdubz** [comment]
So basically I have to train the decoder first. Then warmstart agap training with unfreeze modules = "all"?

**Comment by rafaelvalle** [comment]
That's one option. agap will be trained from scratch because we haven't shared a pre-trained agap yet. 
with dap, you can train jointly the decoder and the feature predictor.
